### PR TITLE
Added: Black formatting verification for whole project

### PR DESCRIPTION
**Pull request by ssbarnea**
Adopts black formatting and runs it as part of the linting process. This should assure that code style is assured by black and we do not have to care about it anymore.

Fixes:  #1305

### CODE REVIEW

File: cookiecutter/generate.py [hunk 1]
```diff
@@ -263,17 +267,10 @@ def generate_files(repo_dir, context=None, output_dir='.',
 
     unrendered_dir = os.path.split(template_dir)[1]
     ensure_dir_is_templated(unrendered_dir)
-    env = StrictEnvironment(
-        context=context,
-        keep_trailing_newline=True,
-    )
+    env = StrictEnvironment(context=context, keep_trailing_newline=True,)
```
Comment (insspb) [inline cookiecutter/generate.py:270]: Trailing commas in such places should be removed

Comment (ssbarnea) [reply cookiecutter/generate.py]: @inglesp I tried to remove the but some of them are added back by black itself.

Comment (inglesp) [reply cookiecutter/generate.py]: I think you meant @insspb...

**Comment by ssbarnea** [comment]
@inglesp Can you please comment again? Github reports a comment from you which seems impossible to find.

**Comment by inglesp** [comment]
> @inglesp Can you please comment again? Github reports a comment from you which seems impossible to find.

image

**Comment by ssbarnea** [comment]
Thanks! That GH but is really annoying, the "hidden comments". I think last version got the commas sorted.

**Pull request merged**